extensions/output/api.js

```diff
@@ -0,0 +1,34 @@
+
+
+module.exports = function container (get) {
+  let c = get('conf')
+  let express = require('express')
+  let app = express()
+  let random_port = require('random-port')
+
+  let run = function(reporter, tradeObject) {
+    if (!reporter.port || reporter.port === 0) {
+      random_port({from: 20000}, function(port) {
+        startServer(port, tradeObject)
```

Comment:
try/catch if the port doesn't work, and we try a different one automatically?

Reply:
okay I will do that

Reply:
the random_port function already searches for a free port, so I dont think this is neccesary anymore. The function also tests explicitly if the port is available through a try / catch mechanism

Reply:
![image](https://user-images.githubusercontent.com/1412006/33567105-6292694e-d922-11e7-8ef6-6657e049eaca.png)

Reply:
Ah, sweet! Thanks for digging that up.